kobo/apps/stripe/tests/test_one_time_addons_tracking.py

```diff
@@ -0,0 +1,196 @@
+        self,
+    ):
+        product = baker.make(
+            Product,
+            active=True,
+            metadata={
+                'product_type': 'plan',
+                'plan_type': 'enterprise',
+                'organizations': True,
+                'asr_seconds_limit': self.subscription_limit,
+            },
+        )
+        price = baker.make(
+            Price,
+            active=True,
+            id='price_sfmOFe33rfsfd36685657',
+            recurring={'interval': 'month'},
+            product=product,
+        )
+
+        subscription = baker.make(
+            Subscription,
+            customer=self.customer,
+            status='active',
+            livemode=False,
+            billing_cycle_anchor=self.now - relativedelta(weeks=2),
+            current_period_end=self.now + relativedelta(weeks=2),
+            current_period_start=self.now - relativedelta(weeks=2),
+        )
+        baker.make(
+            SubscriptionItem,
+            subscription=subscription,
+            price=price,
+            quantity=1,
+            livemode=False,
+        )
+
+    def _create_payment(
+        self, payment_status='succeeded', refunded=False, quantity=1
+    ):
+        metadata = {
+            'product_type': 'addon_onetime',
+            'asr_seconds_limit': self.addon_limit,
+            'valid_tags': 'all',
+        }
+        product = baker.make(
+            Product,
+            active=True,
+            metadata=metadata,
+        )
+        price = baker.make(Price, active=True, product=product, type='one_time')
+        product.save()
+        payment_total = quantity * 2000
+        payment_intent = baker.make(
+            PaymentIntent,
+            customer=self.customer,
+            status=payment_status,
+            payment_method_types=['card'],
+            livemode=False,
+            amount=payment_total,
+            amount_capturable=payment_total,
+            amount_received=payment_total,
+        )
+        self.charge = baker.prepare(
+            Charge,
+            customer=self.customer,
+            refunded=refunded,
+            created=timezone.now(),
+            payment_intent=payment_intent,
+            paid=True,
+            status=payment_status,
+            livemode=False,
+            amount_refunded=0 if refunded else payment_total,
+            amount=payment_total,
+        )
+        self.charge.metadata = {
+            'price_id': price.id,
+            'organization_id': self.organization.id,
+            'quantity': quantity,
+            **(product.metadata),
+        }
+        self.charge.save()
+
+    def test_increment_addon_usage(self):
+        # update_nlp_counter will error out if cache isn't updated first
+        # Also note that the used asr_seconds_current_month has to be > subscription limit,
+        # otherwise the addon will not be incremented (and neither will the counter, therefore)
+        self.organization.update_usage_cache(
```

Comment:
As I mention in the note in the code above, `update_nlp_counter()` does not work for asr/mt usage types if the org usage cache hasn't been updated previously. Unless I'm mistaken, the only time this cache gets updated currently is when the org usage endpoint is accessed. Should we initialize the org usage fields with data that can be used in the process of running `update_nlp_counter()` or should this function be changed to not depend only on this cache data?

Reply:
Planning to address this issue in a subsequent PR.